neqo-transport/tests/connection.rs

```diff
@@ -132,6 +132,70 @@ fn reorder_server_initial() {
     assert_eq!(*client.state(), State::Confirmed);
 }
 
+/// Receiving an ACK frame for a packet number that was never sent is an error.
+#[test]
+fn ack_for_unsent() {
```

Comment:
A driveby for this one...  Did you consider just injecting an ACK frame into the packet?  That might have been easier than this, which is pretty significant in size.  It's only ACK + LA=100000 + 0 + 0.

Reply:
You mean via frame_writer?

Reply:
Right.  `test_frame_writer`.